code/game/machinery/suit_storage_unit.dm

```diff
@@ -309,24 +311,34 @@
 
 	if(uv)
 		if(uv_super)
-			add_overlay("[base_icon_state]_uvstrong")
+			alert_lights = icon(icon, "[base_icon_state]_uvstrong")
 			add_overlay("[base_icon_state]_super")
 		else
-			add_overlay("[base_icon_state]_uv")
+			alert_lights = icon(icon, "[base_icon_state]_uv")
 		add_overlay("[base_icon_state]_lights_red")
+		if(occupant)
+			alert_lights = icon(icon, "[base_icon_state]_body")
+		add_overlay(alert_lights)
 		return
 
 	if(state_open)
 		add_overlay("[base_icon_state]_open")
-		add_overlay("[base_icon_state]_ready")
+		alert_lights = icon(icon, "[base_icon_state]_ready")
+		if(occupant)
+			alert_lights = icon(icon, "[base_icon_state]_body")
+		add_overlay(alert_lights)
```

Comment:
I don't like the new way this is being handled with the `alert_lights` var. The `icon()` proc `new`s an icon every time it's called. But, it's possible that that icon is not even used, and another one is `new`'d further down the chain.

I suggest doing away with `alert_lights` and either just adding the overlay or not.